sdk/maintenance/Azure.ResourceManager.Maintenance/src/Custom/Models/MaintenanceConfigurationData.Serialization.cs

```diff
@@ -12,6 +10,11 @@
 using Azure.ResourceManager.Maintenance.Models;
 using Azure.ResourceManager.Models;
 
+// Custom code to support custom datetime format for the 2 properties startDateTime and expirationDateTime
+// Here is an issue https://github.com/Azure/autorest.csharp/issues/3184 to track if codegen can support custom datetime format in the furture.
+
+[assembly: CodeGenSuppressType("MaintenanceConfigurationData")]
```

Comment:
We should be able to only replace the deserilaization method instead of suppress the entrie class
In addition, the CodeGenSuppressType attribute will also remove the corresponding model factory for this, therefore we should minimize our changes.

Reply:
Fixed, use `CodeGenSuppress` directly overwrite the serialization method.